pysus/ftp/__init__.py

```diff
@@ -1,436 +1,406 @@
-            return Directory(self.path + path)
-        raise ValueError("Unsupported division")
+    def _init_regular(self, parent_path: str, name: str) -> None:
+        self.parent = Directory(parent_path)
+        self.name = name
+        self.loaded = False
+        self.__content__ = {}
 
     @property
-    def content(self):
-        """
-        Returns a list of Files and Directories in the Directory, will load
-        if needed
-        """
+    def content(self) -> List[Union[Directory, File]]:
         if not self.loaded:
             self.load()
         return list(self.__content__.values())
 
-    def load(self):
-        """
-        The content of a Directory must be explicitly loaded
-        """
-        self.__content__ |= load_path(self.path)
+    def load(self) -> Self:
+        self.__content__ |= load_directory_content(self.path)
         self.loaded = True
         return self
 
-    def reload(self):
-        """
-        Reloads the content of the Directory
-        """
-        self.loaded = False
-        return self.load()
-
-    def is_parent(self, other: Union[Self, File]) -> bool:
-        """
-        Checks if Directory or File is inside (or at any subdir) of self.  
-        """
-        if self.path == "/":
-            return True
-
-        target = other
-        while target.path != "/":
-
-            if self.path == target.path:
-                return True
-
-            if isinstance(other, File):
-                # TODO: Implement parent logic on File (too much overhead)
-                target = Directory(other.parent_path)
-            else:
-                target = target.parent
-
-        return False
-
-
-CACHE["/"] = Directory("/")
 
-
-def load_path(path: str) -> Dict[str, Union[Directory, File]]:
-    """
-    This method is responsible for listing all the FTP directory's.
-    Converts the items found within a valid DATASUS path into `File`s or
-    Directories, returning its content.
-    """
-    path = str(path)
-    content = {}
-    ftp = FTP("ftp.datasus.gov.br")
+def load_directory_content(path: str) -> FileContent:
+    """Improved directory content loading with better error handling"""
+    content: FileContent = {}
 
     try:
-        ftp.connect()
-        ftp.login()
+        ftp = FTPSingleton.get_instance()
         ftp.cwd(path)
 
-        def line_file_parser(file_line):
-            info = {}
-            if "<DIR>" in file_line:
-                date, time, _, *name = str(file_line).strip().split()
-                info["size"] = 0
-                info["type"] = "dir"
-                name = " ".join(name)
-                modify = datetime.strptime(
-                    " ".join([date, time]), "%m-%d-%y %I:%M%p"
-                )
-                info["modify"] = modify
-                xpath = (
-                    path + name if path.endswith("/") else path + "/" + name
-                )
-                content[name] = Directory(xpath)
```

Comment:
This line is responsible for populating the Directory's content and should not be removed, otherwise Directory will never load Directories inside its `__content__`